pkg/autoscaler/autoscaler.go

```diff
@@ -254,295 +255,412 @@ var _ = Describe("[Feature:Machines] Autoscaler should", func() {
-				*ms.Spec.Replicas, expectedReplicas))
+				return *ms.Spec.Replicas == expectedReplicas
+			}, framework.WaitLong, pollingInterval).Should(BeTrue())
+		})
 
-			return *ms.Spec.Replicas == expectedReplicas
-		}, framework.WaitMedium, pollingInterval).Should(BeTrue())
+		It("cleanup deletion information after scale down [Slow]", func() {
+			By("Creating 2 MachineSets each with 1 replica")
+			var transientMachineSets [2]*mapiv1beta1.MachineSet
+			targetedNodeLabel := fmt.Sprintf("%v-delete-cleanup", autoscalerWorkerNodeRoleLabel)
+			for i, machineSet := range transientMachineSets {
+				machineSetParams := framework.BuildMachineSetParams(client, 1)
+				machineSetParams.Labels[targetedNodeLabel] = ""
+				machineSet, err = framework.CreateMachineSet(client, machineSetParams)
+				Expect(err).ToNot(HaveOccurred())
+				cleanupObjects[machineSet.GetName()] = machineSet
+				transientMachineSets[i] = machineSet
+			}
 
-		By("Waiting for the machineSet replicas to become nodes")
-		framework.WaitForMachineSet(client, machineSet.GetName())
+			By("Waiting for all Machines in MachineSets to enter Running phase")
+			framework.WaitForMachineSet(client, transientMachineSets[0].GetName())
+			framework.WaitForMachineSet(client, transientMachineSets[1].GetName())
+
+			expectedReplicas := int32(3)
+			for _, machineSet := range transientMachineSets {
+				By(fmt.Sprintf("Creating a MachineAutoscaler backed by MachineSet %s - min: 1, max: %d",
+					machineSet.GetName(), expectedReplicas))
+				asr := machineAutoscalerResource(machineSet, 1, expectedReplicas)
+				Expect(client.Create(ctx, asr)).Should(Succeed())
+				cleanupObjects[asr.GetName()] = asr
+			}
 
-		expectedReplicas = 0
-		By("Deleting the workload")
-		Expect(deleteObject(workload.Name, cleanupObjects[workload.Name])).Should(Succeed())
-		delete(cleanupObjects, workload.Name)
-		Eventually(func() bool {
-			ms, err := framework.GetMachineSet(client, machineSet.GetName())
-			Expect(err).ToNot(HaveOccurred())
+			jobReplicas := expectedReplicas * int32(2)
+			By(fmt.Sprintf("Creating scale-out workload: jobs: %v, memory: %s",
+				jobReplicas, workloadMemRequest.String()))
+			workload := newWorkLoad(jobReplicas, workloadMemRequest, targetedNodeLabel)
+			cleanupObjects[workload.GetName()] = workload
+			Expect(client.Create(ctx, workload)).Should(Succeed())
 
-			By(fmt.Sprintf("Waiting for machineSet replicas to scale in. Current replicas are %v, expected %v.",
-				*ms.Spec.Replicas, expectedReplicas))
+			for _, machineSet := range transientMachineSets {
+				By(fmt.Sprintf("Waiting for MachineSet %s replicas to scale out", machineSet.GetName()))
+			}
+			Eventually(func() (bool, error) {
+				sizeCount := 0
+				for _, machineSet := range transientMachineSets {
+					current, err := framework.GetMachineSet(client, machineSet.GetName())
+					if err != nil {
+						return false, err
+					}
+					if *current.Spec.Replicas == expectedReplicas {
+						sizeCount++
+					}
```

Comment:
Could invert this equality and return early, then you don't need `sizeCount`
```suggestion
					if *current.Spec.Replicas != expectedReplicas {
						return false, nil
					}
```

Reply:
good spot, this is a leftover from when i had the logs printing inside the loop. i'll remove it.